GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -443,11 +456,35 @@ + (GIDSignIn *)sharedInstance {
   static dispatch_once_t once;
   static GIDSignIn *sharedInstance;
   dispatch_once(&once, ^{
-    sharedInstance = [[self alloc] initPrivate];
+    GTMKeychainStore *keychainStore =
+        [[GTMKeychainStore alloc] initWithItemName:kGTMAppAuthKeychainName];
+    sharedInstance = [[self alloc] initWithKeychainStore:keychainStore];
   });
   return sharedInstance;
 }
 
+#pragma mark - Configuring and pre-warming
+
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+- (void)configureWithCompletion:(nullable void (^)(NSError * _Nullable))completion {
+  @synchronized(self) {
+    // This check helps to avoid needing to set `_appCheck` in an initializer called in
+    // `+[GIDSignIn sharedInstance]`. Also defers to the fakes for `_appCheck` set in tests.
+    // Ultimately, we would prefer to use `GIDSignIn`'s initializer to pass this information in, but
+    // this is currently impossible given that `GIDSignIn` is a singleton.
+    // TODO: Remove this once `GIDSignIn` is no longer a singleton (https://github.com/google/GoogleSignIn-iOS/issues/322)
```

Comment:
I'm not sure this is accurate. Although `GIDSignIn` is a singleton at the public API level, it's been a non-singleton in our internal implementation and tests, because we can use the private initializer `initWithKeychainStore:`. What is blocking us from using `initWithKeychainStore:appCheckProvider:` for the singleton created in `sharedInstance` in this PR?

Reply:
If we move creating `_appCheck` to `+[GIDSignIn sharedInstance]`, then we'll have something like this:

```
+ (GIDSignIn *)sharedInstance {
  static dispatch_once_t once;
  static GIDSignIn *sharedInstance;
  dispatch_once(&once, ^{
    GTMKeychainStore *keychainStore =
        [[GTMKeychainStore alloc] initWithItemName:kGTMAppAuthKeychainName];
#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
    if (@available(iOS 14.0, *)) {
      sharedInstance = [[self alloc] initWithKeychainStore:keychainStore
                                          appCheckProvider:nil];
    }
#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
    if (!sharedInstance) {
      sharedInstance = [[self alloc] initWithKeychainStore:keychainStore];
    }
  });
  return sharedInstance;
}
```

This is fine, but then we'd need to remove `-[GIDSignInTest testSharedInstance]` since we are currently depending on Firebase proper, which requires that the `FirebaseApp` is configured prior to getting an instance of app check:

```
2023-07-18 16:38:16.089762-0700 xctest[2658:1509258] 10.10.0 - [FirebaseCore][I-COR000003] The default Firebase app has not yet been configured. Add `FirebaseApp.configure()` to your application initialization. This can be done in in the App Delegate's application(_:didFinishLaunchingWithOptions:)` (or the `@main` struct's initializer in SwiftUI). Read more: https://goo.gl/ctyzm8.
2023-07-18 16:38:16.092787-0700 xctest[2658:1508993] *** Terminating app due to uncaught exception 'com.firebase.appCheck', reason: 'The default FirebaseApp instance must be configured before the defaultAppCheck instance can be initialized. One way to ensure this is to call `FirebaseApp.configure()` in the App Delegate's `application(_:didFinishLaunchingWithOptions:)` (or the `@main` struct's initializer in SwiftUI).'
```

I've removed the singleton test since I don't think it does much more than test how singletons are implemented.

Reply:
Thanks! Can you add a TODO in `initWithKeychainStore:appCheckProvider:` to move initialization of `GIDAppCheck` to `sharedInstance` when Firebase App Check doesn't require `FirebaseApp.configure()`?

Reply:
Sorry, I think my reply was confusing. I believe I have addressed this already in [this commit](https://github.com/google/GoogleSignIn-iOS/pull/314/commits/2538f619ca9d3b7ffbbc450671623341f75c411d#diff-b76ad8011382daf05bf7a830db6abe2428c68fbeadd4b617b39bd2c504e2b640R535); however, to add further confusion, I have streamlined my approach in this [new commit](https://github.com/google/GoogleSignIn-iOS/pull/314/commits/74913be3d79b1a3a9f1661206b43ff9399cad136#diff-b76ad8011382daf05bf7a830db6abe2428c68fbeadd4b617b39bd2c504e2b640R463-R467).

Reply:
I.e., no need for a TODO since we can create `GIDAppCheck` within `sharedInstance` if we do not have `-[GIDSignInTest testSharedInstance]`.